src/pages/SingleVideoPage/SingleVideoPage.js

```diff
@@ -1,37 +1,57 @@
 import "./SingleVideoPage.css";
+import {useEffect} from "react";
 import {useData} from "../../contexts";
 import {useParams,Link} from "react-router-dom";
 import {useDocumentTitle} from "../../customhooks";
+import axios from "axios";
 
 function SingleVideoPage(){
-
     const { videos } = useData();
     const { _id } = useParams();
-    const getSingleVideo = videos.filter((item) => item._id===_id);
-    let getSimilarVideos = videos.filter((item) => item.categoryName === getSingleVideo[0].categoryName);
+    const video = videos.filter((item) => item._id===_id)[0];
```

Comment:
You can use find method here which will return the element of the videos array that matches the condition. You can skip the indexing at the end.